app/src/androidTest/java/com/github/houseorganizer/houseorganizer/FirestoreShopListTest.java

```diff
@@ -0,0 +1,123 @@
+package com.github.houseorganizer.houseorganizer;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+
+import com.github.houseorganizer.houseorganizer.shop.FirestoreShopList;
+import com.github.houseorganizer.houseorganizer.shop.ShopItem;
+import com.google.android.gms.tasks.Task;
+import com.google.android.gms.tasks.Tasks;
+import com.google.firebase.firestore.DocumentSnapshot;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.QuerySnapshot;
+
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+@RunWith(AndroidJUnit4.class)
+public class FirestoreShopListTest {
+
+    private static FirestoreShopList localShopList;
+    private static FirebaseFirestore db;
+
+    @BeforeClass
+    public static void startingFirebase() throws ExecutionException, InterruptedException {
+        FirebaseTestsHelper.startAuthEmulator();
+        FirebaseTestsHelper.startFirestoreEmulator();
+        FirebaseTestsHelper.setUpFirebase();
+        db = FirebaseFirestore.getInstance();
+        Task<QuerySnapshot> t = db.collection("shop_lists").get();
+        Tasks.await(t);
+        assertThat(t.getResult().getDocuments().size(), is(1));
```

Comment:
Asserting things in this type of methods can make all the test fails, did you have that in mind while doing it?

Reply:
Nice catch I will make the test less specific but if there are 0 shopLists it is normal that everything fails as I cannot compare the local data to the remote one.